File: flutter_inappwebview_windows/windows/types/save_as_ui_showing_response.cpp
#include "save_as_ui_showing_response.h"

namespace flutter_inappwebview_plugin
{
  SaveAsUIShowingResponse::SaveAsUIShowingResponse(const std::optional<bool>& cancel,
    const std::optional<bool>& suppressDefaultDialog,
    const std::optional<std::string>& saveAsFilePath,
    const std::optional<bool>& allowReplace,
    const std::optional<SaveAsKind>& kind)
    : cancel(cancel), suppressDefaultDialog(suppressDefaultDialog), saveAsFilePath(saveAsFilePath),
    allowReplace(allowReplace), kind(kind)
  {}

  SaveAsUIShowingResponse::SaveAsUIShowingResponse(const flutter::EncodableMap& map)
    : SaveAsUIShowingResponse(get_optional_fl_map_value<bool>(map, "cancel"),
      get_optional_fl_map_value<bool>(map, "suppressDefaultDialog"),
      get_optional_fl_map_value<std::string>(map, "saveAsFilePath"),
      get_optional_fl_map_value<bool>(map, "allowReplace"),
      SaveAsKindFromOptionalInteger(get_optional_fl_map_value<int64_t>(map, "kind")))
  {}

  flutter::EncodableMap SaveAsUIShowingResponse::toEncodableMap() const
  {
    return flutter::EncodableMap{
      {"cancel", make_fl_value(cancel)},
      {"suppressDefaultDialog", make_fl_value(suppressDefaultDialog)},
      {"saveAsFilePath", make_fl_value(saveAsFilePath)},
      {"allowReplace", make_fl_value(allowReplace)},
      {"kind", make_fl_value(SaveAsKindToInteger(kind))}
    };
  }
}

File: flutter_inappwebview_platform_interface/lib/src/types/save_as_ui_showing_response.dart
import 'package:flutter_inappwebview_internal_annotations/flutter_inappwebview_internal_annotations.dart';

import 'save_as_kind.dart';
import 'enum_method.dart';

part 'save_as_ui_showing_response.g.dart';

///Class that represents the response used by the [PlatformWebViewCreationParams.onSaveAsUIShowing] event.
@ExchangeableObject()
class SaveAsUIShowingResponse_ {
  ///Whether to cancel the Save As action.
  bool? cancel;

  ///Whether to suppress the default dialog.
  bool? suppressDefaultDialog;

  ///The save file path to use.
  String? saveAsFilePath;

  ///Whether the save operation may replace an existing file.
  bool? allowReplace;

  ///The save-as kind for the operation.
  SaveAsKind_? kind;

  SaveAsUIShowingResponse_({
    this.cancel,
    this.suppressDefaultDialog,
    this.saveAsFilePath,
    this.allowReplace,
    this.kind,
  });
}
